backend/src/main/java/com/tourneynizer/tourneynizer/dao/TeamRequestDao.java

```diff
@@ -56,31 +56,52 @@ public TeamRequest requestTeam(User user, Team team) {
         return insert(team.getId(), user.getId(), user.getId());
     }
 
-    public void requestUser(User requested, Team team, User requester) {
+    public TeamRequest requestUser(User requested, Team team, User requester) {
         if (team.getCreatorId() != requester.getId()) {
             throw new IllegalArgumentException("Only the creator of a team can request other users");
         }
-        insert(team.getId(), requested.getId(), requester.getId());
+        return insert(team.getId(), requested.getId(), requester.getId());
     }
 
     private final RowMapper<Long> idMapper = (resultSet, i) -> resultSet.getLong(1);
 
-    public List<Long> getRequests(Team team) {
-        String sql = "SELECT user_id FROM teamRequest WHERE team_id=?;";
+    private <T> List<T> getRequestsHelper(RowMapper<T> mapper, String sql, long param1) {
+        return getRequestsHelper(mapper, sql, new Long[]{param1});
+    }
+
+    private <T> List<T> getRequestsHelper(RowMapper<T> mapper, String sql, Long[] params) {
         return this.jdbcTemplate.query(connection -> {
             PreparedStatement preparedStatement = connection.prepareStatement(sql);
-            preparedStatement.setLong(1, team.getId());
+            for (int i = 0; i < params.length; i++) {
+                preparedStatement.setLong(i+1, params[i]);
+            }
             return preparedStatement;
-        }, idMapper);
+        }, mapper);
+    }
+
+    public List<Long> getRequestIds(Team team) {
+        String sql = "SELECT user_id FROM teamRequest WHERE team_id=?;";
+        return getRequestsHelper(idMapper, sql, team.getId());
     }
 
-    public List<Long> getRequests(User user) {
+    public List<Long> getRequestIds(User user) {
         String sql = "SELECT team_id FROM teamRequest WHERE user_id=?;";
-        return this.jdbcTemplate.query(connection -> {
-            PreparedStatement preparedStatement = connection.prepareStatement(sql);
-            preparedStatement.setLong(1, user.getId());
-            return preparedStatement;
-        }, idMapper);
+        return getRequestsHelper(idMapper, sql, user.getId());
+    }
+
+    public List<TeamRequest> getTeamRequests(Team team) {
+        String sql = "SELECT * FROM teamRequest WHERE team_id=? AND requester_id<>?;";
+        return getRequestsHelper(teamRequestRowMapper, sql, new Long[]{team.getId(), team.getCreatorId()});
+    }
```

Comment:
Why not also do a requests for team and by team?

Reply:
That's for another PR. Currently there isn't an endpoint for a team to view its requests.